File: 112-1/Discrete_Mathematics/cashier.js
// denominations
function change(n, d){
	coins = {}
	for(let i of d.sort((a,b) => b - a)){
		while(n >= i){
			coins[i] = coins[i] ? coins[i] + 1 : 1
			n -= i
		}
	}
	console.log(coins)
	return Object.keys(coins).reduce((a, c) => a + c)
}

n = 16469
d = [1, 5, 10, 50, 100, 500, 1000]

change(n, d)

File: 112-1/Discrete_Mathematics/cashier.py
def change(d, n):
    count = 0
    for i in d:
        while(n >= i):
            count += 1
            n -= i
    return count 

n = 63
d = [1000, 500, 100, 50, 10, 5, 1]

print(change(d, n))

